Read me PT-BR

matheuscassiano opened the pull request:
[![Build Status](https://travis-ci.org/EventFahrplan/EventFahrplan.svg?branch=master)](https://travis-ci.org/EventFahrplan/EventFahrplan) [![Apache License](http://img.shields.io/badge/license-Apache%20License%202.0-lightgrey.svg)](http://choosealicense.com/licenses/apache-2.0/)

# EventFahrplan

EventFahrplan é um visualizador para agendas de eventos, por exemplo
[Chaos Computer Club e.V. events][ccc-events] como o Chaos Communication Camp
ou Congresso de Comunicação do Caos.

O aplicativo está disponível para dispositivos Android.

Relatórios de bugs, solicitações de recursos podem ser enviadas como um [issue][issue].
Por favor, leia as instruções no [contribution guide](CONTRIBUTING.md) para contribuir com este projeto.

![Picture of the 35c3 Schedule on tablet and phone](gfx/EventFahrplan-35c3-Events-Tablet-Phone.png)

## Índice

- [Recursos](#features)
- [Dados de eventos](#event-data)
- [Instruções de build](#build-instructions)
- [Historico](#history)
- [Licenças](#licenses)

## Recursos

* Resumo diário dos eventos
* Leia resumos, descrições de eventos
* Adicione eventos à lista de favoritos
* Exportar lista de favoritos
* Configurar alarmes para eventos individuais
* Adicione eventos ao seu calendário pessoal
* Compartilhe um link para um evento com outras pessoas
* Idiomas suportados: holandês, inglês, francês, alemão, italiano, português, russo (descrições de eventos excluídas)

## Dados de evento

* O aplicativo foi projetado para consumir dados de eventos publicados em um formato específico
conforme fornecido por [Frab][frab-github]. O formato do arquivo produzido pelo predecessor do software, [Pentabarf][pentabarf-github],não pode ser consumido imediatamente.
* Em geral, é possível reimplementar o aplicativo para outros eventos que
forneça o mesmo formato de arquivo de evento.

## Instruções de build

Antes de importar este projeto em Android Studio.

Na pasta `app`, faça uma cópia do arquivo` gradle.properties.example` e renomeie-o como `gradle.properties`.
Este arquivo contém as informações de assinatura usadas no processo de build.

## Historico

* O projeto foi iniciado como ["CampFahrplan"][campfahrplan-github] em 2011 e foi desenvolvido
por [Daniel Dorau][tuxmobil-github]. Ele publicou o aplicativo para o Chaos Communication Camp
e Congresso da Comunicação do Caos nos anos seguintes. O aplicativo serviu como um digital
agendar para milhares de usuários.
* Em 2013, [Tobias Preuss][johnjohndoe-github] começou a contribuir. Logo depois ele
começou a reimplementar o aplicativo para outros eventos, como FOSSGIS, FrOSCon, MRMCD e
outras conferências.
* Em agosto de 2017, o projeto mudou-se para um novo local e foi renomeado para
["EventFahrplan"][eventfahrplan-github] reconhecer seu uso mais amplo.


## Licenças

Portions Copyright 2008-2011 The K-9 Dog Walkers and 2006-2011 the Android Open Source Project.


```
Copyright 2013-2019 johnjohndoe
Copyright 2011-2015 Daniel Dorau
Contributions from Adriano Pereira Junior, Akarsh Seggemu, Andrea Marziali,
Andy Scherzinger, Andreas Schildbach, Animesh Verma, bashtian, bjoernb,
ButterflyOfFire, cacarrara, Caio Volpato, cketti, codingcatgirl, entropynil,
koelnkalkverbot, ligi, Mateus Baptista, MichaelRocks, Noemis, Poschi,
Sjors van Mierlo, Stefan Medack, SubOptimal, Torsten Grote, Vladimir Alabov

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
```

[campfahrplan-github]: https://github.com/tuxmobil/CampFahrplan
[ccc-events]: http://events.ccc.de
[eventfahrplan-github]: https://github.com/EventFahrplan/EventFahrplan
[frab-github]: https://github.com/frab/frab
[johnjohndoe-github]: https://github.com/johnjohndoe
[pentabarf-github]: https://github.com/nevs/pentabarf
[tuxmobil-github]: https://github.com/tuxmobil/CampFahrplan
[issue]: https://github.com/EventFahrplan/EventFahrplan/issues

johnjohndoe committed:
Use Android plugin v.3.5.0.

+ Changelog: https://developer.android.com/studio/releases/gradle-plugin#3-5-0.

johnjohndoe committed:
Disallow release candidates as upgradeable versions.

+ Configures the gradle-versions-plugin.

johnjohndoe committed:
Merge pull request #178 from EventFahrplan/imgbot

[ImgBot] Optimize images

johnjohndoe committed:
Merge branch 'project-setup'

johnjohndoe committed:
Let TalkBack read out language abbreviations as full words.

johnjohndoe committed:
Merge pull request #179 from EventFahrplan/accessibility-talkback-language-abbreviation

Let TalkBack read out language abbreviations as full words.

Commit:
Spanish translation complete.

adrianoapj committed:
Portuguese updated and defined as a built-in language.

Commit:
Add Russian language

AniVerma17 committed:
Limits the app activities to be displayed over lock screen

johnjohndoe committed:
Let code become more testable by injecting a Moment.

johnjohndoe committed:
Merge pull request #183 from today-/master

Add Russian language

johnjohndoe committed:
Update list of contributors.

sjorsvanmierlo committed:
created the dutch main strings.xml file

AniVerma17 committed:
Fixes favorites fragment visibility over lock screen on tablets

johnjohndoe committed:
Apply automatic formatting.

johnjohndoe committed:
Merge branch 'portuguese-translation'

johnjohndoe committed:
Merge pull request #186 from spmvanmierlo/feature/DutchTranslations

Feature/dutch translations

johnjohndoe committed:
Update list of contributors.

Commit:
Add italian translations

johnjohndoe committed:
Merge pull request #187 from amarziali/add-italian-translation

Add italian translations

johnjohndoe committed:
Update list of contributors.

johnjohndoe committed:
Merge pull request #184 from AniVerma17/lockscreen-fix

Limits the app activities to be displayed over lock screen

johnjohndoe committed:
Update list of contributors.

johnjohndoe committed:
Update list of supported languages.

yanicka committed:
Add russian translation

+ Completed the two optional tasks

johnjohndoe commented:
Hello @matheuscassiano. I appreciate your effort. I however don't see a good reason to have the README being translated into multiple languages. The source code is English, therefore programmers need to be capable of understanding English in any case. Can you please state your motivation?

johnjohndoe committed:
Merge pull request #190 from vherasme/spanish-translation

Spanish translation

johnjohndoe committed:
Update list of supported languages.

johnjohndoe committed:
Update list of contributors.

johnjohndoe committed:
Merge pull request #191 from yanicka/add-russian-translation

Add russian translation

johnjohndoe committed:
Update list of contributors.

johnjohndoe committed:
Fix minor typo.

+ Since 21176763d8b3f5e480c1d909e2b9089e087f5564, relates to #190.

johnjohndoe committed:
Link sponsor profile in this repository.

johnjohndoe committed:
Facilitate easier testing by injecting AppRepository and passing it along.

johnjohndoe committed:
Use AppRepository to access the "last_fetched_at" preference.

+ The value of the preference key changes here. Although this
  might look like a breaking change it means that a user has
  to fetch data one more time in the worst case.

johnjohndoe committed:
Don't pass :network:FetchScheduleResult further then needed.

johnjohndoe committed:
Extract "parseSchedule" method.

johnjohndoe committed:
Create and manage OkHttpClient outside of AppRepository.

johnjohndoe committed:
Extract "showErrorDialog" method.

johnjohndoe committed:
Use AppRepository to access the "changes_seen" preference.

+ Related commit: 91b8e6bdc2046fcf7581ae01cf418325b2ff0c54.

johnjohndoe committed:
Remove unused import.

johnjohndoe committed:
Add preference item to enter Engelsystem URL.

johnjohndoe committed:
Validate Engelsystem URL entered by the user.

johnjohndoe committed:
Wrap parsing results into "ParseScheduleResult" class.

johnjohndoe committed:
Ensure that fetching and parsing the schedule is consumed.

johnjohndoe committed:
Improve formatting.

johnjohndoe committed:
Encapsulate methods to load lectures and highlights within AppRepository.

johnjohndoe committed:
Load shifts from Engelsystem API.

+ If configured for the product flavor the app will fetch
  shift data from the Engelsystem backend and parse it.
+ The shift data is converted into the data structure which
  is used for schedule data.
+ Any shift which exceeds the time range of the conference
  days as defined by the schedule data is dropped.
+ New shift data replaces data loaded earlier.
+ A separate room/column is shown on a particular day if
  shift data is available.
+ This commit does not implement UI to enter an API URL.
+ Texts of the feature are translated to English and German.

johnjohndoe committed:
Add missing override annotations.

johnjohndoe committed:
Merge branch 'housekeeping'

johnjohndoe committed:
Merge pull request #196 from EventFahrplan/revise-architecture

Revise architecture around AppRepository

johnjohndoe committed:
Use gradle-versions-plugin v.0.27.0.

+ Changelog: https://github.com/ben-manes/gradle-versions-plugin/releases/tag/v0.27.0.

johnjohndoe committed:
Use sonarqube-gradle-plugin v.2.8.

johnjohndoe committed:
Merge branch 'dependencies'

johnjohndoe committed:
Use Gradle wrapper v.5.6.4.

+ Changelog: https://docs.gradle.org/5.6.4/release-notes.html.

johnjohndoe committed:
Use Android plugin v.3.5.2.

+ Changelog: https://developer.android.com/studio/releases/gradle-plugin#3-5-0.

johnjohndoe committed:
Merge branch 'project-setup'

johnjohndoe committed:
Initialize AppRepository once in application class.

+ This avoids that each component needs to pass Context to AppRepository.

johnjohndoe committed:
Merge pull request #197 from EventFahrplan/apprepository

Initialize AppRepository once in application class.

johnjohndoe committed:
Fix NullPointerException when trying to delete an alarm list item.

+ Related commits: 6a957359dc7a0be79c47a186ab01365b17892eea,
  141a981f3c677bb66f63100ab61633a38b244a19.

johnjohndoe committed:
Return the Unit object instead of null where it is appropriate.

johnjohndoe committed:
Use email-intent-builder v.2.0.0.

+ Changelog: https://github.com/cketti/EmailIntentBuilder#changelog.

johnjohndoe committed:
Use okhttp v.3.12.6.

+ Changelog: https://square.github.io/okhttp/changelog_3x/#version-3126.

johnjohndoe committed:
Use mockito v.3.2.0.

+ Changelog: https://github.com/mockito/mockito/blob/release/3.x/doc/release-notes/official.md#320.

johnjohndoe committed:
Use Kotlin v.1.3.61.

+ Changelog: https://github.com/JetBrains/kotlin/releases/tag/v1.3.61.

akarsh committed:
Add time and place

- Encapsulate loading vector drawables to render without crash on Android 4 (API 14-19)
- Add icon for time and place
-  Add vector assets for the two image files
     - ic prefix is used before the both image file names,
       - room
       - access time
 - Add supportVectorDrawables
- Change fillColor for drawables
- Add new dimen for event detail bar icon padding
- Add new colours to change drawable colours
- Add support for drawables in gradle file
- Add new placeholder for the id - event location
  - Refactor placeholder value for the id - event date
  - Add value to the place id - event location

Co-authored-by: Tobias Preuss <tobias.preuss@googlemail.com>

johnjohndoe committed:
Merge pull request #192 from akarsh/Add_icon_for_time_and_place

Add icon for time and place

johnjohndoe committed:
Merge branch 'unit-housekeeping'

johnjohndoe committed:
Merge branch 'dependencies'

johnjohndoe committed:
Let the operating system check SSL/TLS certificates.

+ Drop custom SSLSocketFactory.
+ Release build successfully tested on:
  - Pixel 2 device, Android 10 (API 29)
  - Nexus 5 device, Android 6.0.1 (API 23)
  - Nexus 5 emulator, Android 5.0.2 (API 21)
  - Nexus 5 emulator, Android 4.4.2 (API 19)
  - Nexus 5X emulator, Android 4.1.2 (API 16)

johnjohndoe committed:
Clarify the scope of the string resources.

johnjohndoe committed:
Prepare for handling different parse results.

+ Add ParseResult interface.

johnjohndoe committed:
Apply automatic formatting to strings.xml files.

johnjohndoe committed:
Remove unused imports.

johnjohndoe committed:
Facilitate testing of AppRepository.

johnjohndoe committed:
Merge pull request #198 from EventFahrplan/ssl-socket-factory

Let the operating system check SSL/TLS certificates.

johnjohndoe committed:
Merge branch 'housekeeping'

johnjohndoe committed:
Use Android plugin v.3.5.3.

+ Changelog: https://developer.android.com/studio/releases/gradle-plugin#3-5-0.

johnjohndoe committed:
Customize user agent to let the APIs identify requests made by the app.

johnjohndoe committed:
Merge pull request #199 from EventFahrplan/user-agent

Customize user agent to let the APIs identify requests made by the app.

johnjohndoe committed:
Document room index calculation.

johnjohndoe committed:
Add configuration for 36C3.

johnjohndoe committed:
Add resources for 36C3.

johnjohndoe committed:
Update c3nav integration for 36C3.

johnjohndoe committed:
Clear database from 35C3 on upgrade.

+ Related commit: 2aab57a51dc96c00aa53248c5edfec179f4132e9.

johnjohndoe committed:
Remove unused color resources.

+ Unused since commit 8cd6681bdaa405abbaa3f58db0b7ff709ba99867
  resp. commit cea61b4e2382b32d869601d09b312cb44342c85e.

johnjohndoe committed:
Merge branch 'colors-housekeeping'

johnjohndoe committed:
Define the conference name once so changing it for the c3nav snack is easy.

johnjohndoe committed:
Merge branch 'c3nav-housekeeping'

johnjohndoe committed:
Merge pull request #200 from EventFahrplan/engelsystem-shifts

Integrate Engelsystem shifts

johnjohndoe committed:
Remove unused parameter.

+ Left-over from commit 6a957359dc7a0be79c47a186ab01365b17892eea.

johnjohndoe committed:
Merge branch 'housekeeping'

johnjohndoe committed:
Prepare "Engelshifts" to be colorized based on their track name.

+ Former information was used for debugging.
+ Related commit: e6cc6486a8818514060fec8c2fdd30b9f0d39d7b.

johnjohndoe committed:
Customize color for Engelsystem shifts for 36C3.

+ Related commit: 7becad223fda7742987f42b40b01a8d3438cf564.

johnjohndoe committed:
Merge pull request #201 from EventFahrplan/36c3

36c3

johnjohndoe committed:
Update "Features" section.

johnjohndoe committed:
Merge branch 'features-documentation'

johnjohndoe committed:
Bump application version to v.1.41.0.

johnjohndoe committed:
Fix NullPointerException in UpdateService#onDestroy.

+ Thanks to @schildbach + @cketti for their support.

Resolves #205.

johnjohndoe committed:
Merge pull request #206 from EventFahrplan/NullPointerException-UpdateService-onDestroy

Fix NullPointerException in UpdateService#onDestroy.

johnjohndoe committed:
Update room mappings for c3nav.

+ Based on https://github.com/c3nav/c3nav/blob/80abc62825c134ea8ea1ec99bf94c71c1d8783a2/res/mappings.36c3.json.

johnjohndoe committed:
Merge pull request #207 from EventFahrplan/c3nav-room-mappings

Update room mappings for c3nav.

johnjohndoe committed:
Bump application version to v.1.41.1.

johnjohndoe committed:
Unit test Lecture#sanitize extension.

+ Related commit: 39f9eba1332d428e3ecb2211a9a7feb28786f2d1.
+ Related issue: #13.

johnjohndoe committed:
Clear "subtitle" field if it matches the "speakers" names.

+ Related commit: 39f9eba1332d428e3ecb2211a9a7feb28786f2d1.
+ Related issue: #13.

johnjohndoe committed:
Sanitize lecturized Engelsystem shifts.

johnjohndoe committed:
Automatically reload schedule if Engelsystem shifts URL is configured.

+ Related commit: 7bb735c7bcc2b0ec29b9e93c7116c6801005d9a4.

johnjohndoe committed:
Merge pull request #209 from EventFahrplan/engelsystem-auto-reload-schedule

Automatically reload schedule if Engelsystem shifts URL is configured.

cketti committed:
Only use signingConfig when relevant properties are set

+ Update build instructions in README

johnjohndoe committed:
Merge pull request #208 from EventFahrplan/lecture-housekeeping

Sanitize lectures

johnjohndoe committed:
Bump application version to v.1.41.2.

cketti committed:
Don't use MyApp.lectureList to populate EventDetailFragment

The MyApp.lectureList mechanism is populating Lecture.isHighlight and
Lecture.hasAlarm. AppRepository.readLectureByLectureId() was therefore
updated to also provide this functionality.

Resolves #204

cketti committed:
Avoid creating lots of temporary objects

The column index logic in MetaDatabaseRepository wasn't doing anything
useful. So it was removed.

johnjohndoe committed:
Update room mapping for c3nav.

+ Based on https://github.com/c3nav/c3nav/blob/01befdedb82c96d1bf99f3ac88d0dccb833145ba/res/mappings.36c3.json.
+ Related commit: 4a15fb11beef1997195a0cd8d473b5a1321360fc.

johnjohndoe committed:
Merge pull request #217 from EventFahrplan/36c3-room-mappings

Update room mappings for c3nav.

johnjohndoe committed:
Facilitate more verbose user feedback.

+ Translations for other languages are missing!

johnjohndoe committed:
Merge pull request #219 from EventFahrplan/crash-reporting-message

Facilitate more verbose user feedback.

johnjohndoe committed:
Merge pull request #214 from cketti/fix_EventDetailFragment

Don't use MyApp.lectureList to populate EventDetailFragment

johnjohndoe committed:
Bump application version to v.1.41.3.

ideadapt committed:
#125 room title on single line with padding and ellipsis if overflow.

johnjohndoe committed:
Mitigates the effect of a SecurityException which can occur in JobIntentService.

+ Addresses the error:
  "SecurityException: Caller no longer running, last stopped".

Resolves #120.

johnjohndoe committed:
Ensure de/selecting a favorite in details is reflected in the main view.

+ The main schedule view relies on the correct state of the static
  lecture list (MyApp.lectureList) in order to render the favor state.
  Therefore, it is required that once a lecture is modified in the details
  view also the corresponding entry in the static lecture list (if present)
  is updated.
+ Bug introduced in 8d0107331a4d88dc2626d5095cbf3349d0d66a00.
+ Relates to issue #204, #214.

johnjohndoe committed:
Merge pull request #223 from EventFahrplan/update-main-lecture-view

Ensure de/selecting a favorite in details is reflected in the main view.

johnjohndoe committed:
Merge pull request #221 from EventFahrplan/SecurityException_JobIntentService

Mitigates the effect of a SecurityException which can occur in JobIntentService.

johnjohndoe committed:
Merge pull request #220 from ideadapt/bugfix-125/room-title

Fix long room titles

johnjohndoe committed:
Update list of contributors.

johnjohndoe committed:
Bump application version to v.1.41.4.

ideadapt committed:
#195 update notification channels on locale changed event by simply recreating (overriding) them.

ideadapt committed:
Start consolidating different date handling helpers, utils etc.

Refactor Moment internals to use classes of threeten backport library.
ThreeTenBP is required if minSdkVersion <= 25.
In Java 8 a new java.time implementation was shipped, aka threeten.
This new implementation is only available if minSdkVersion >= 26.

The aim is to have:
a) less strings representing dates (but classes)
b) an own Moment implementation that is UTC based for all calculation stuff.
   Only it might be possible to retrieve fields such as year, month, day
   and time in system time zone.
c) distinct class for date formatting

See also: #92, https://arturdryomov.dev/posts/modern-times

---

Failing tests:

+ LectureExtensionsTest
+ NavigationMenuEntriesGeneratorTest
+ DateInfoTest

ideadapt committed:
Move DateHelper start/end of day methods to Moment.

Remove zoneOffset from DayRange:

* For comparing dates in DayRange.contains it's not required (ZoneDateTime compares based on epoch millis)
* For logging purposes UTC suits well.

See also: #92

---

Failing tests:

+ ShiftExtensionsTest

Fixes:

+ LectureExtensionsTest
+ NavigationMenuEntriesGeneratorTest
+ DateInfoTest

ideadapt committed:
Fix DateHelper.getMinuteOfDay(long) and getMinuteOfDay(ZonedDateTime) which did not respect zone offset.

E.g. ZonedDateTime of "2019-01-02T02:00.000GMT+1" is the same as "2019-01-02T01:00.000Z",
but getMinuteOfDay returned 120 and 60 respectively. Now its fixed, hence 60 for both inputs.

See also: #92

---

Failing tests:

+ TimeSegmentTest

Fixes:

+ ShiftExtensionsTest

johnjohndoe committed:
Update copyright notices.

johnjohndoe committed:
Structure preferences XML as done for other languages.

johnjohndoe committed:
Reorganize string resources for "Alternative schedule URL" preference.

johnjohndoe committed:
Use AppRepository as a common interface to access PREFS_SCHEDULE_URL.

johnjohndoe committed:
Automatically reload schedule if alternative schedule URL is configured.

+ Related commit: 43c6b6c2659a20df3d02cfd8c1084a5d96e5add4.

johnjohndoe committed:
Allow to exclude "Alternative schedule URL" preference from build.

+ Configuring an alternative schedule URL does not make sense for every
  event where the app is used. The retrieved schedule must be compatible
  with what the app expects aka. the original schedule.
  The preference was initially added as a backup if the main backend is
  not reachable for a longer time during an event.
+ Related commit: d387a4f8f4e9ea17328416cbc72d9c2202494a55.

johnjohndoe committed:
Merge pull request #233 from EventFahrplan/preferences-alternative-schedule-url

Preferences alternative schedule url

johnjohndoe committed:
Hide feedback button for wiki events because there is no feedback system.

+ This only works as long as the "track" name in the "schedule.xml" is
  set to "self organized sessions" for these sessions.
  Let's see how long this works ..

johnjohndoe committed:
Hide "Event online" section if the events URL is empty.

johnjohndoe committed:
Prevent composing invalid URLs for Engelsystem shifts.

+ If a shift contains an URL then it is displayed in the "Events online"
  section.
+ This stops the "EventUrlComposer" from creating an URL based on the
  ID of the shift leading to an invalid URL.

johnjohndoe committed:
Bugfix: Stop displaying alien URLs in Engelsystem shifts.

+ This fixes an URL being shown in the "Event online" section of an
  Engelsystem shift which is associated with a different event.
+ The bug was caused by an overlap of the database IDs ("event_id" column).

johnjohndoe committed:
Merge pull request #234 from EventFahrplan/wiki-events-no-feedback

Hide feedback button for wiki events because there is no feedback system.

johnjohndoe committed:
Merge pull request #235 from EventFahrplan/event-url-composition

Stop displaying alien URLs in Engelsystem shifts

johnjohndoe committed:
Clear "subtitle" if it matches the "title" of a session.

johnjohndoe committed:
Merge pull request #236 from EventFahrplan/identical-title-subtitle

Clear "subtitle" if it matches the "title" of a session.

johnjohndoe committed:
Allow to use logging in all modules.

+ Introduce "commons" module.
+ Debug logging is no longer controlled by the MyApp.LogDebug flag.

johnjohndoe committed:
Merge pull request #237 from EventFahrplan/logging-all-modules

Allow to use logging in all modules.

johnjohndoe committed:
Record debug information about zombie lecture which sometimes crashes the app.

+ Follow-up of 8d0107331a4d88dc2626d5095cbf3349d0d66a00.
+ Relates to #238.

johnjohndoe committed:
Merge pull request #239 from EventFahrplan/NoSuchElementException-logging

Record debug information about zombie lecture which sometimes crashes the app.

johnjohndoe committed:
Use Gradle wrapper v.6.0.1.

+ Changelogs:
  https://docs.gradle.org/6.0/release-notes.html
  https://docs.gradle.org/6.0.1/release-notes.html

johnjohndoe committed:
Merge branch 'project-setup'

johnjohndoe committed:
Use engelsystem v.3.1.0.

+ Changelog: https://github.com/johnjohndoe/engelsystem/blob/master/CHANGELOG.md#v310.

+ Updates in the dependency tree:

  info.metadude.kotlin.library.engelsystem:engelsystem-base:3.1.0
    -> com.squareup.okhttp3:okhttp:3.12.7
  com.squareup.okhttp3:okhttp:3.12.6 -> 3.12.7

johnjohndoe committed:
Use okhttp v.3.12.7.

+ Changelog: https://square.github.io/okhttp/changelog_3x/#version-3127.

+ Updates in the dependency tree:

  info.metadude.kotlin.library.engelsystem:engelsystem-base:3.1.0
    -> com.squareup.okhttp3:okhttp:3.12.7
  com.squareup.okhttp3:okhttp:3.12.7

johnjohndoe committed:
RIP 35C3.

+ Update screenshot.

johnjohndoe committed:
Merge pull request #240 from EventFahrplan/rip-35c3

RIP 35C3.

ideadapt committed:
Moment now operates in UTC time zone.

Before field getters where based on system default timezone.
UTC based values are timezone neutral and therefore good for comparison & calculation purposes.

See also: #92

ideadapt committed:
Implement Moment.isBefore.

See also: #92

johnjohndoe committed:
Merge pull request #215 from cketti/born_to_copy

Clean up LecturesDatabaseRepository

johnjohndoe committed:
Merge pull request #210 from cketti/signingConfigs

Only use signingConfig when relevant properties are set

johnjohndoe committed:
Merge branch 'dependencies'

johnjohndoe committed:
Prevent crashes caused by closed database connections.

+ Errors:
  + SQLiteDatabaseLockedException: database is locked.
  + IllegalStateException: Cannot perform this operation
    because the connection pool has been closed.
+ Crashes might still occur since AlarmList uses a CursorAdapter
  to access and manage the database.
+ Related commit: acfef066bfc83a602c2faf507b56e8b914be12e9

Resolves #229, #228.

johnjohndoe committed:
Link to installable applications which have been built from this project.

+ Relates to #188.

johnjohndoe committed:
Merge pull request #242 from EventFahrplan/SQLiteOpenHelper-close

Prevent crashes caused by closed database connections.

johnjohndoe committed:
Bump application version to v.1.41.5.

johnjohndoe committed:
Update contribution guide.

cketti committed:
Clean up reading from Cursor

ideadapt committed:
Use DateHelper to format lecture detail time in EventDetailFragment.

Use static fields for formatters to avoid repeated instance creation.

See also: #92

ideadapt committed:
Comment relStartTime and startTime semantics of Lecture model.

Group semantically related fields.

See also: #92

ideadapt committed:
Refactor Moment internals and add new constructor.

* Improve endOfDay calculation by doing less custom arithmetic, but using API constants.
* Delegate DateHelper.getMinuteOfDay calculation to Moment.
* Use injection friendly Instant construction (org.threeten.bp.Clock.systemUTC() could be mocked in tests).

ideadapt committed:
Migrate DateFieldValidation to Kotlin. Refactor to use existing date helper classes.

* Move congress.utils.DateHelper to commons.DateHelper
* Remove now unused methods from network.utils.DateHelper.kt.

See also: #92

ideadapt committed:
Move network.DateHelper content to Moment, DateFormatter and DateParser.

network.util.DateHelper defined redundant functionality of different concerns:
parsing, formatting and calculation. All but parsing is now extracted
and DateHelper renamed to DateParser.
commons.DateHelper has been reduced to formatting functionality in previous
commits and is now called commons.DateFormatter.

DateHelper#getDayChange has required a fallback of 600 minutes.
This fallback is not required anymore since its replacement method DateParser#getDateTime
is more robust.
Please note that the very same default value is still used in ParserTask#parseFahrplan
because the corresponding <day_change> XML node is only present in the
schedule.xml file when retrieved from a Pentabarf API.

See also: #92

johnjohndoe committed:
Merge pull request #225 from ideadapt/bugfix-195/notification-channel-locale

#195 update notification channels on locale changed event

ideadapt committed:
Make DateFormatter none static. More comparable date formats in DateFieldValidation error message.

DateFormatter static fields were introduced in commit (Use DateHelper to format ...). This decision did not prove successful.
Static is hard to mock. The nullable fields lead to NullPointerExceptions now and then.

See also: #92

ideadapt committed:
Use zone aware date formatter to render TimeSegment.

TimeSegment is used to render vertical time series at the left side in the day view.
It was not time zone aware. Now it is. Time is always rendered in devices default time zone.
So far, this is the only place a date is rendered in devices timezone.
Maybe we should consider to always display dates in the conferences timezone?

See also: #92

---

Fixes:

- TimeSegmentTest

johnjohndoe committed:
Merge pull request #244 from cketti/cursor_cleanup

Clean up reading from Cursor

johnjohndoe committed:
Use truth v.1.0.1.

+ Changelog: https://github.com/google/truth/releases/tag/release_1_0_1.

johnjohndoe committed:
Use junit v.4.13.

+ Changelog: https://github.com/junit-team/junit4/blob/master/doc/ReleaseNotes4.13.md.

johnjohndoe committed:
Use unmock v.0.7.4.

+ Changelog: https://github.com/bjoernQ/unmock-plugin#versions.

johnjohndoe committed:
Use threetenbp v.1.4.1.

+ Changelog: https://www.threeten.org/threetenbp/changes-report.html#a1.4.1.

johnjohndoe committed:
Use okhttp v.3.12.8.

+ Changelog: https://square.github.io/okhttp/changelog_3x/#version-3128.

johnjohndoe committed:
Merge branch 'dependencies'

johnjohndoe committed:
Use build tools v.29.0.3.

johnjohndoe committed:
Use Gradle wrapper v.6.1.1.

+ Changelogs:
  https://docs.gradle.org/6.1/release-notes.html
  https://docs.gradle.org/6.1.1/release-notes.html

johnjohndoe committed:
Merge branch 'project-setup'

ideadapt committed:
Use existing DateParser to parse date String instead of threeten Instant

See also #92

johnjohndoe committed:
Do not render "Schedule " prefix when schedule version is not provided.

+ Context: Some APIs do not provide any version information with the
  schedule.xml. The leads to nothing but the word "Schedule " being shown
  in the relevant part of the "About" screen.

ideadapt committed:
Implement Moment.plusMinutes. Replaced Moment.plusSeconds(xMinutes * 60).

See also #92

ideadapt committed:
Expand all `import java.util.*` imports.

See also #92

ideadapt committed:
Use expression body syntax where suitable.

See also #92

ideadapt committed:
Add KDoc to new public API.

See also #92

ideadapt committed:
Migrate Moment class to a data class.

See also #92

johnjohndoe committed:
Use gradle-versions-plugin v.0.28.0.

+ Note: the Gradle Versions Plugin now requires at least Gradle 5.0 to function.
+ Changelog: https://github.com/ben-manes/gradle-versions-plugin/releases/tag/v0.28.0.
+ Prepare for Gradle v.6.2.
+ Update resolution strategy.

johnjohndoe committed:
Use Gradle wrapper v.6.2.2.

+ Changelogs:
  https://docs.gradle.org/6.2/release-notes.html
  https://docs.gradle.org/6.2.1/release-notes.html#fixed-issues
  https://docs.gradle.org/6.2.2/release-notes.html#fixed-issues

johnjohndoe committed:
Use unmock v.0.7.5.

+ Changelog: https://github.com/bjoernQ/unmock-plugin#versions.

johnjohndoe committed:
Use okhttp v.3.12.10.

+ Changelog: https://square.github.io/okhttp/changelog_3x/#version-31210.

johnjohndoe committed:
Use mockito v.3.3.0.

+ Changelog: https://github.com/mockito/mockito/blob/release/3.x/doc/release-notes/official.md#330.

johnjohndoe committed:
Merge pull request #247 from EventFahrplan/about-screen-schedule-version

Do not render "Schedule " prefix when schedule version is not provided.

johnjohndoe committed:
Merge branch 'project-setup'

johnjohndoe committed:
Encapsulate loadChangedLectures() within AppRepository.

johnjohndoe committed:
Encapsulate loadStarredLectures() within AppRepository.

johnjohndoe committed:
Improve misleading method names in database repositories.

johnjohndoe committed:
Let database delete all favored lectures.

+ Actually, the "highlight" column is set to 0 for all rows in the
  "highlight" table in the "highlight" database. None of the rows is
  really removed - they are just flagged.
+ While MyApp.lectureList is still around it needs to be manually
  updated to stay in sync with the database state.

johnjohndoe committed:
Make debugging more pleasant.

johnjohndoe committed:
Use string template.

johnjohndoe committed:
Merge pull request #245 from ideadapt/bugfix-92/spread-date-lib-usage

Centralize date handling

johnjohndoe committed:
Improve logcat output.

+ Consistently use class names without package names
  because they are cut off.

johnjohndoe committed:
Inline variable.

johnjohndoe committed:
Merge branch 'housekeeping'

johnjohndoe committed:
Encapsulate loadUncanceledLecturesForDayIndex() within AppRepository.

johnjohndoe committed:
Unit test AppRepository#updateLecturesLegacy.

+ Related commit: c2d45668c35c0d6ad9ac86c552a2ac6436170f11.

johnjohndoe committed:
Use existing AppRepository#updateLecturesLegacy to reset "highlight" flag.

+ While MyApp.lectureList is still around it needs to be manually
  updated to stay in sync with the database state.

johnjohndoe committed:
Unit test AlarmServices.

+ Document public API.

ideadapt committed:
Merge pull request #249 from EventFahrplan/observing-data-part-01

Observing data - part 01

johnjohndoe commented:
@matheuscassiano Hello. Please check out https://github.com/EventFahrplan/EventFahrplan where the development of the project has moved to.

johnjohndoe committed:
Remove dead code.

+ Unused since its creation, see e6cc6486a8818514060fec8c2fdd30b9f0d39d7b.

johnjohndoe committed:
Encapsulate change calculations in new ChangeStatistic class.

johnjohndoe committed:
Remove EVENT_TIME bundle extra which is never read.

+ Related commit: fedaf930cade93b0b0cb907eb93d940cfcbfa7dd.

johnjohndoe committed:
Remove redundant null check.

johnjohndoe committed:
Use camelcase instead of snakecase to comply with Java naming convention.

johnjohndoe committed:
Remove unused field.

+ Unused since 8c195fd4c4181227db84020104406c8602302bf6.

johnjohndoe committed:
Use primitive type in boolean expression.

+ Works the same in ChangeListFragment and StarredListFragment.

johnjohndoe committed:
Remove redundant semicolon.

johnjohndoe committed:
Convert into expression body.

johnjohndoe committed:
Avoid misleading variable name.

johnjohndoe committed:
Remove redundant initialization.

johnjohndoe committed:
Use standard library function.

johnjohndoe committed:
Add missing annotation.

johnjohndoe committed:
Remove unused throws expression from method signature.

johnjohndoe committed:
Mark declaration as final.

johnjohndoe committed:
Suppress false positive Lint warning regarding constant value.

+ The value is derived from flavor specific build configuration.

johnjohndoe committed:
Access static reference.

johnjohndoe committed:
Prefer interfaces over specific implementations.

johnjohndoe committed:
Collapse empty XML element.

johnjohndoe committed:
Remove duplicates from .gitignore file.

johnjohndoe committed:
Add TODO to refactor date processing.

johnjohndoe committed:
Remove auto-generated TODOs.

johnjohndoe committed:
Remove useless override method.

johnjohndoe committed:
Fix null check.

+ Related commit: 8e6eeeacd029875f4ae2b1dc48569995dbe399ec.

ideadapt committed:
Merge pull request #252 from EventFahrplan/change-statistic

Encapsulate change calculations in new ChangeStatistic class.

johnjohndoe committed:
Customize the toolbar overflow menu style.

johnjohndoe committed:
Clean up logging output for alarm scheduling.

johnjohndoe committed:
Customize the context menu style.

+ The menu which pops up when long-pressing a lecture.

johnjohndoe committed:
Customize the spinner dropdown item style used in the day chooser.

johnjohndoe committed:
Add visual touch feedback to the spinner style used in the alarm time picker.

+ This affects ways to launch the alarm time picker:
  - via the context menu of a lecture in the schedule screen
  - via the toolbar action in the lecture details screen
+ Both android:spinnerItemStyle and android:spinnerDropDownItemStyle must
  be configured in Theme.Congress resp. Theme.Congress.NoActionBar because
  AlarmTimePickerFragment#initializeSpinner relied on the system layouts
  android.R.layout.simple_spinner_item resp.
  android.R.layout.simple_spinner_dropdown_item

johnjohndoe committed:
Maintain time column colors in a central location.

ideadapt committed:
Merge pull request #253 from EventFahrplan/alarm-services

Unit test AlarmServices.

johnjohndoe committed:
Maintain room name header colors in a central location.

+ Improve rendering of layout preview.

johnjohndoe committed:
Prefix color resource to group with related resources.

ideadapt committed:
Merge pull request #255 from EventFahrplan/housekeeping

Random housekeeping

ideadapt committed:
Merge pull request #257 from EventFahrplan/customizing-menues-and-colors

Customizing menues and colors
